eva-remapping-ingest/src/test/java/uk/ac/ebi/eva/remapping/ingest/batch/io/RemappedSubmittedVariantsWriterTest.java

```diff
@@ -0,0 +1,194 @@
+        hashingFunction = new SubmittedVariantSummaryFunction().andThen(new SHA1HashingFunction());
+    }
+
+    @After
+    public void tearDown() {
+        mongoClient.dropDatabase(TEST_DB);
+        remappingIngestCounts.resetCounts();
+    }
+
+    private SubmittedVariantEntity createSve(Long accession, long start, String ref, String alt,
+                                             LocalDateTime createdDate, String remappedFrom) {
+        SubmittedVariant model = new SubmittedVariant("GCA_000000002.1", 1000, "projectId_1", "CM000002.1",
+                                                      start, ref, alt, 3000000002L);
+        String hash = hashingFunction.apply(model);
+        SubmittedVariantEntity sve = new SubmittedVariantEntity(accession, hash, model, 1);
+        sve.setCreatedDate(createdDate);
+        sve.setRemappedFrom(remappedFrom);
+        return sve;
+    }
+
+    private SubmittedVariantEntity createSveInAssembly(String assembly,
+                                                       Long accession, long start, String ref, String alt,
+                                                       LocalDateTime createdDate, String remappedFrom) {
+        SubmittedVariant model = new SubmittedVariant(assembly, 1000, "projectId_1", "CM000002.1",
+                                                      start, ref, alt, 3000000002L);
+        String hash = hashingFunction.apply(model);
+        SubmittedVariantEntity sve = new SubmittedVariantEntity(accession, hash, model, 1);
+        sve.setCreatedDate(createdDate);
+        sve.setRemappedFrom(remappedFrom);
+        return sve;
+    }
+
+    private void assertRemappingIngestCounts(int ingested, int skipped, int discarded) {
+        assertEquals(ingested, remappingIngestCounts.getRemappedVariantsIngested());
+        assertEquals(skipped, remappingIngestCounts.getRemappedVariantsSkipped());
+        assertEquals(discarded, remappingIngestCounts.getRemappedVariantsDiscarded());
+    }
+
+    @Test
+    public void testDoesNotWriteNewDuplicateAccessions_discardFromInput() {
+        // Ensure the source variant we'd find is later than whatever is already in the db
+        SubmittedVariantEntity sourceSve = createSveInAssembly("GCA_000000001.5", 5000000002L, 2100, "C", "T",
+                                                               LocalDateTime.now(), null);
+        mongoTemplate.insert(sourceSve);
+
+        List<SubmittedVariantEntity> svesToWrite = Collections.singletonList(
+                createSve(5000000002L, 2100, "C", "T", LocalDateTime.now(), "GCA_000000001.5"));
+        writer.write(svesToWrite);
+        assertRemappingIngestCounts(0, 0, 1);
+    }
+
+    @Test
+    public void testDoesNotWriteNewDuplicateAccessions_discardFromDatabase() {
+        List<SubmittedVariantEntity> svesToWrite = Collections.singletonList(
+                createSve(5000000002L, 2100, "C", "T", LocalDateTime.now(), null));
+        writer.write(svesToWrite);
```

Comment:
I don't see an insert into the DB with the colliding SVE??

Reply:
You're absolutely correct... thank you!